devlib/utils/android.py

```diff
@@ -294,6 +294,10 @@ def __init__(
             total_transfer_timeout=total_transfer_timeout,
             transfer_poll_period=transfer_poll_period,
         )
+
+        self.logger.debug('AdbConnection(): server=%s port=%s device=%s as_root=%s',
```

Comment:
There is a slight technical benefit of using this way of interpolating (i.e. avoiding interpolation unless that debug level is enabled) but it's negligible and mostly just increases the risk of mixup. You can use an f-string instead to avoid that, and even let the compiler avoid redundancy with e.g.  `{adb_server=}`.